### PR TITLE
LUN-99: Fix result screen bugs

### PR DESCRIPTION
**Pull request by SteffiStuffel**
This pull request belongs to an issue on our [bugtracker](https://issues.tuerantuer.org/secure/RapidBoard.jspa?rapidView=27&view=detail).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword _LUN_.

### CODE REVIEW

File: src/screens/ResultScreen.tsx [hunk 1]
```diff
@@ -124,7 +132,7 @@ const ResultScreen = ({ route, navigation }: ResultScreenPropsType): JSX.Element
 
   const Footer = (
     <>
-      {retryButton}
+      {exercise === ExerciseKeys.vocabularyTrainer && retryButton}
```
Comment (klinzo) [inline src/screens/ResultScreen.tsx:135]: Why do we only have a retry button for the vocabulary trainer?

File: src/screens/ResultScreen.tsx [hunk 2]
```diff
@@ -72,10 +72,18 @@ interface ResultScreenPropsType {
...
   const { Icon, title, order } = resultType
-  const nextResultType = RESULTS.find(result => result.order === (order + 1) % RESULTS.length) ?? RESULTS[0]
+
+  let nextResultType = RESULTS.find(result => result.order === (order + 1) % RESULTS.length) ?? RESULTS[0]
+  if (
+    nextResultType.key === 'similar' &&
+    (exercise === ExerciseKeys.learnArticles || exercise === ExerciseKeys.singleChoice)
+  ) {
+    nextResultType = RESULTS[2]
+  }
```
Comment (klinzo) [inline src/screens/ResultScreen.tsx:86]: What is this doing? :thinking:

File: src/screens/ResultScreen.tsx (hunk 1 above)
Comment (SteffiStuffel) [reply src/screens/ResultScreen.tsx:135]: I guess, the writeExercise is considered to be more complex and therefore has the opportunity to repeat the wrong entries. Maybe in the future, we will add this for singleChoice exercises, too, if it seems necessary.

File: src/screens/ResultScreen.tsx (hunk 2 above)
Comment (SteffiStuffel) [reply src/screens/ResultScreen.tsx:86]: For single choice exercises there is no option for "almost correct" results, and therefore they are skipped.

**Review by klinzo — APPROVED** [review]

**Pull request merged**